contracts/MechMarketplace.sol

```diff
@@ -469,9 +634,8 @@ contract MechMarketplace is IErrorsMarketplace {
             revert ZeroValue();
         }
 
-        // Check mech validity
-        bool status = IMech(mech).checkMechValidity(mech);
-        if (!status) {
+        // Check mech validity as it must be created and recorded via this marketplace
+        if (mapAgentMechFactories[mech] == address(0)) {
```

Comment:
If the mech was not recorded by this marketplace, it must not be allowed to proceed.